File: c++ Data Structure/stack.cpp
#include<iostream>
using namespace std;
int position =0,stack[100];
int main()
{
	int ch=0;
	while(ch<4)
	{
		cout<<"1.push 2.pop 3.display ";
		cin>>ch;
		switch(ch)
		{
			case 1:
			{
				if(position==100)
				{
					cout<<"Stack full"<<endl;
				}
				else
				{
					cout<<"Enter value to be entered ";
					cin>>stack[position];
					position++;
				}
				break;
			}
			case 2:
			{
				if(position==0)
				{
					cout<<"Stack is empty"<<endl;
				}
				else
				{
					position--;
					cout<<"popped"<<endl;
				}
				break;
			}
			case 3:
			{
				int i;
				for(i=0;i<position;i++)
				{
					cout<<stack[i]<<endl;
				}
				break;
			}
		}
	}
}

/* output: 

1.push 2.pop 3.display 1
Enter value to be entered 10
1.push 2.pop 3.display 1
Enter value to be entered 20
1.push 2.pop 3.display 1
Enter value to be entered 30
1.push 2.pop 3.display 3
10
20
30
1.push 2.pop 3.display 2
popped
1.push 2.pop 3.display 3
10
20
1.push 2.pop 3.display 5

*/

File: Python Data Structure/stack.py
class Node:
    def __init__(self):
        self.data=None
        self.next=None
        self.head=None
    def push(self,data):
        if(self.head==None):
            newnode=Node()
            newnode.data=data
            self.head=newnode
        else:
            node=self.head
            while(node.next is not None):
                node=node.next
            newnode=Node()
            newnode.data=data
            node.next=newnode
    def display(self):
        node=self.head
        while(node is not None):
            print(node.data)
            node=node.next
    def pop(self):
        node=self.head
        while(node.next.next is not None):
            node=node.next
        del(node.next)
        node.next=None


obj=Node()
ch=0
while(ch<4):
    print("1.push 2.pop 3.display")
    ch=int(input())
    if(ch==1):
        print("enter element")
        element=int(input())
        obj.push(element)
    elif(ch==2):
        print("element popped")
        obj.pop()
    elif(ch==3):
        print("Display stack")
        obj.display()
